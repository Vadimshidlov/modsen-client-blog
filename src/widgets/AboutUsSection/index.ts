export { AboutUsSection } from '@/widgets/AboutUsSection/ui/AboutUsSection';
