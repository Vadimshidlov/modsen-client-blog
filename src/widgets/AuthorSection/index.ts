export { AuthorSection } from '@/widgets/AuthorSection/ui/AuthorSection';
