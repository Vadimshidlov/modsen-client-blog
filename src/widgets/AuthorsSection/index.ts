export { AuthorsSection } from '@/widgets/AuthorsSection/ui/AuthorsSection';
