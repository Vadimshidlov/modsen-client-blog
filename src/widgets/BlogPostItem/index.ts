export { BlogPostItem } from '@/widgets/BlogPostItem/BlogPostItem';
