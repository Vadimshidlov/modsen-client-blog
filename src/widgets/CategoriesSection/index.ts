export { CategoriesSection } from '@/widgets/CategoriesSection/ui/CategoriesSection';
