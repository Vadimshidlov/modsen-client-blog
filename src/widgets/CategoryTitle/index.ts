export { CategoryTitle } from '@/widgets/CategoryTitle/ui/CategoryTitle';
