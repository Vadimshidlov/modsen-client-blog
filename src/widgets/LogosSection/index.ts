export { LogosSection } from '@/widgets/LogosSection/ui/LogosSection';
