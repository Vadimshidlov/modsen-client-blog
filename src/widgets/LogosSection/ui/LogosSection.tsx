import React from 'react';
import Link from 'next/link';
import Image from 'next/image';
import { useTranslations } from 'next-intl';
import { LOGOS_LIST } from '@/widgets/LogosSection/constants/logos';
import container from '@/shared/styles/container.module.scss';
import styles from '@/widgets/LogosSection/ui/LogosSection.module.scss';

export function LogosSection() {
  const t = useTranslations('LogosSection');

  return (
    <section className={`${styles.logosSection} ${container.container}`}>
      <div className={styles.logosTitleContainer}>
        <span className={styles.logosSubTitle}>{t('title')}</span>
        <span className={styles.logosTitle}>{t('subTitle')}</span>
      </div>
      <ul className={styles.logosContainer}>
        {LOGOS_LIST.map(({ id, img, link }) => (
          <li key={id}>
            <Link href={link}>
              <Image src={img} alt="logo-pic" />
            </Link>
          </li>
        ))}
      </ul>
    </section>
  );
}
