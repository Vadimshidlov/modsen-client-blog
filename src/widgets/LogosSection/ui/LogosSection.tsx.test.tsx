import React from 'react';
import { render } from '@testing-library/react';
import { LogosSection } from '@/widgets/LogosSection';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('LogosSection tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<LogosSection />);
    expect(container).toMatchSnapshot();
  });
});
