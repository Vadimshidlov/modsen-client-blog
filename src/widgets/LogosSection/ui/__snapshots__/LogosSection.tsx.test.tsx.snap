// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogosSection tests should matches snapshot 1`] = `
<div>
  <section
    class="logosSection container"
  >
    <div
      class="logosTitleContainer"
    >
      <span
        class="logosSubTitle"
      >
        Mocked title
      </span>
      <span
        class="logosTitle"
      >
        Mocked subTitle
      </span>
    </div>
    <ul
      class="logosContainer"
    >
      <li>
        <a
          href="/"
        >
          <img
            alt="logo-pic"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          <img
            alt="logo-pic"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          <img
            alt="logo-pic"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          <img
            alt="logo-pic"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          <img
            alt="logo-pic"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
      </li>
    </ul>
  </section>
</div>
`;
